### PR TITLE
Support Play 2.6.x for Scala 2.11 as well #20

### PR DESCRIPTION
**Pull request by noamBarkai**

### CODE REVIEW

**Comment by dragisak** [comment]
@noamBarkai I don't think this PR ads 2.6 support for 2.11. Play 2.6 with scala 2.11 should already be supported.

Play versions you see in build.sbt are the lowest common denominators. So this plugins should work for any play version > 2.4 and any version of Scala from 2.10 to 2.12. It's just that 2.6 is the first version to support 2.12. Play version is not pulled transitively by this plugin as the dependency is set as "provided".

I am tempted to close this PR.

**Comment by noamBarkai** [comment]
@dragisak note that prior to this PR (see: play-liquibase/build.sbt) , play-liquibase with Scala 2.11 would assume Play framework 2.4.6 and whereas Scala 2.12 version of this plug in would assume Play 2.6.0
And indeed you are right in writing that Play 2.6.5 supports both Scala 2.11 and Scala 2.12.
Hence the change introduced in this PR - if accepted, then play-liquibase would support Play 2.6.5 for both Scala versions.
Without this PR, one cannot use play-liquibase with Scala 2.11 and Play 2.6.5. The reason one cannot do it is not that Play 2.6.5 doesn't support Scala 2.11 - it does. It is rather that play-liquibase assumes in Scala 2.11 version that it is using Play 2.4.6 - and these two versions of _Play_ - 2.4.6 and 2.6.5 - are not compatible. When trying to do so you get the error I've attached to the original issue.
Bottom line: I think this PR does what it says it will do and adds real value to the plugin - I suggest not closing the PR but rather merging it :)

**Comment by dragisak** [comment]
@noamBarkai I understand what's your concern. That's why play dependency in the [pom file](https://repo1.maven.org/maven2/com/ticketfly/play-liquibase_2.11/1.4/play-liquibase_2.11-1.4.pom) is set to scope "provided". 

That means that play-liquibase does not choose which version of play you will use. `"com.typesafe.play" %% "play"` will be chosen by your project.  

See more in https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html

**Comment by noamBarkai** [comment]
Dragisa, it's not just a concern, it's a real issue. This library simply
does not work as is with Play 2.6 and Scala 2.11
If you're not convinced you're welcome to try, you'll get the error i added
to the issue.
Even though Play is 'provided', still Scala compiler creates a call to
logger.info() with the wrong signiture, so at runtime when the jvm tries to
find it, it fails giving rise to the runtime exception.
My PR fixes this issue for us. I think it's worthwhile to add it back to
the oss project, but it's your call in the end.
Hope this helps

**Comment by noamBarkai** [comment]
BTW, if you want to see the root cause, take a look at Play 2.4.x logger.info()
https://github.com/playframework/playframework/blob/2.4.x/framework/src/play/src/main/scala/play/api/Logger.scala#L95
Compare that to Play 2.6.x logger.info()
https://github.com/playframework/playframework/blob/2.6.x/framework/src/play/src/main/scala/play/api/Logger.scala#L150
You'll easily see the function's signature has changed (they've added an implicit MarkerContext)
Now, play-liquibase in Scala 2.11 assumes Play 2.4 - so the code expects to find logger.info with a single parameter. But if what's actually provided is Play 2.6 then it won't find that method - it now expects another implicit parameter - hence the exception.
I'm rock solid about this. I think it's _very_ clear.
Your call.

**Comment by dragisak** [comment]
@noamBarkai Thanks. It was unclear from the description that the problem was in the change of logging API in 2.6.x. We should probably add separate tests for different major play versions. Right now, it's only testing 2.5.

I'll see if I can get this merged. I left Ticketfly a long time ago and they removed me from maintainers list. If nothing works, I'll fork this repo.
